src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -903,17 +903,7 @@ private async Task StartRequestAsync(WinHttpRequestState state)
                 ThrowOnInvalidHandle(connectHandle, nameof(Interop.WinHttp.WinHttpConnect));
                 connectHandle.SetParentHandle(_sessionHandle);
 
-                // Try to use the requested version if a known/supported version was explicitly requested.
-                // Otherwise, we simply use winhttp's default.
-                string? httpVersion = null;
-                if (state.RequestMessage.Version == HttpVersion.Version10)
-                {
-                    httpVersion = "HTTP/1.0";
-                }
-                else if (state.RequestMessage.Version == HttpVersion.Version11)
-                {
-                    httpVersion = "HTTP/1.1";
-                }
+                string httpVersion = $"HTTP/{state.RequestMessage.Version.Major}.{state.RequestMessage.Version.Minor}";
```

Comment:
note that this change causes a (minor) allocation of the string per each request.

Reply:
This PR contains some experiments for the CI machines atm, will get back to this once my plate is not full anymore